docs/content/Theming/03_configuring_a_theme.md

```diff
@@ -132,3 +132,28 @@ angular.module('myApp', ['ngMaterial'])
 
 });
 </hljs>
+
+### Under the Hood
+
+Angular Material dynamically generates CSS for registered themes by injecting several
+`<script>` tags into the `<head>` section of the application at runtime. Here is how
+that process currently works:
+
+1. During the Angular Material build process, a function in `gulp/util.js` called
+[themeBuildStream](https://github.com/angular/material/blob/master/gulp/util.js#L187)
```

Comment:
The files I'm referring to are in each component directory (e.g. `autocomplete-theme.scss`, `backdrop-theme.scss`, `bottom-sheet-theme.scss`).